src/Zicht/Bundle/UrlBundle/Url/ShortUrlManager.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Zicht Online <http://zicht.nl>
+ */
+
+namespace Zicht\Bundle\UrlBundle\Url;
+
+use Zicht\Bundle\UrlBundle\Aliasing\Aliasing;
+use Zicht\Bundle\UrlBundle\Entity\UrlAlias;
+
+// phpcs:disable Zicht.NamingConventions.Functions.NestedDefinition
+
+class ShortUrlManager
+{
+    /**
+     * @var ShortUrlGeneratorInterface
+     */
+    private $shortUrlGenerator;
+
+    /**
+     * @var Aliasing
+     */
+    private $aliasing;
+
+    /**
+     * @param Aliasing $aliasing
+     * @param ShortUrlGeneratorInterface|null $shortUrlGenerator
+     */
+    public function __construct(Aliasing $aliasing, ShortUrlGeneratorInterface $shortUrlGenerator = null)
+    {
+        $this->aliasing = $aliasing;
+        $this->shortUrlGenerator = $shortUrlGenerator ?: new class() implements ShortUrlGeneratorInterface {
+            /**
+             * {@inheritDoc}
+             */
+            public function generate($url, $length = 8)
+            {
+                return substr(hash('sha1', $url), 0, $length);
+            }
+        };
```

Comment:
Ik zou het netter vinden om deze code in een aparte PHP file te zetten (bijv. `Sha1HashShortUrlGenerator`). En dan hier `new Sha1HashShortUrlGenerator()` aanroepen.

Reply:
Mja, daar heb ik nou juist een keer niet voor gekozen :smile: er is maar 1 hele simpele methode die geimplementeerd hoeft te worden met 1 regel er in :shrug:

Reply:
@boudewijn-zicht wat denk jij?

Reply:
Ben het eens met Erik.  Lijkt me niet nodig om een losse file te maken om een hash te genereren

Reply:
Kun je er dan niet gewoon een callback van maken? Ik vind het nu wel echt een heel lelijke oplossing

Reply:
Daarmee dwing je dan weer geen argumenten af en dat is geen DI-contract

Reply:
Ok, 2 tegen 1... Ik :heavy_check_mark: wel :smile: